types/mongoose/index.d.ts

```diff
@@ -2551,6 +2551,7 @@ declare module "mongoose" {
      * @param fn optional callback
      */
     save(fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
+    save(options: SaveOptions, fn?: (err: any, product: this, numAffected: number) => void): Promise<this>;
```

Comment:
It wouldn't show errors, but try typing:
```js
doc.save({
  va
```
The autocomplete should look like this:
<img width="498" alt="screen shot 2017-04-07 at 7 24 04 pm" src="https://cloud.githubusercontent.com/assets/10586163/24824825/dc386f96-1bc7-11e7-86dd-199b187e4496.png">

If it looks like this:
<img width="511" alt="screen shot 2017-04-07 at 7 25 18 pm" src="https://cloud.githubusercontent.com/assets/10586163/24824833/f82bc838-1bc7-11e7-8897-eb14954f1744.png">

Then it's not detecting that the options object is type `SaveOptions`. Does it work the same way on your computer?